### DISCUSSION TITLE
How to print a SQL query without executing it

### DISCUSSION THREAD

**Discussion by vladinator1000** [prisma/prisma#9626] [Q&A] [answered]
Hi y'all, is there a way to print the query without sending it to the database? I'm working on a tool that allows the creation of "data migrations". There's a related discussion in #4688, but I want to take it in a different direction.

I want to run a script that compiles client code into .sql into the format that prisma migrate expects. Here's what I have so far, I successfully print the SQL but how do I prevent it from going to the database? I tried [not calling next in a middleward](https://www.prisma.io/docs/concepts/components/prisma-client/middleware/logging-middleware) but the query printing stopped working.

```ts
import { PrismaClient } from '../prisma'

const prisma = new PrismaClient({
  log: [
    {
      emit: 'event',
      level: 'query',
    },
  ],
})

prisma.$on('query', async (e) => {
  // print to .sql file
  console.log(`${e.query} ${e.params}`)
})

export async function createDataMigration() {
  await prisma.user.create({
    data: {
      name: 'Hello',
      password: 'hello',
    },
  })
}

createDataMigration()
  .catch((e) => {
    throw e
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

```

**Comment by ryands17** [comment, marked as answer]
@vladinator1000 👋 

There's no way to just print a query without executing it at the moment in Prisma. You would need to create a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

There's a similar 3rd party package that does something like this that you might want to check out: https://github.com/lirancr/prisma-rays

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
